### PR TITLE
Recursion in DeviceListViewModel.UpdateConnectedDevices

### PR DESCRIPTION
**Pull request by alex-g-axion**
- Avoid recursion when showing toast messages on the main thread

Fixes https://axionbio.jira.com/browse/NAV-6747

### CODE REVIEW

File: Src/ToastNotifications/Display/NotificationsDisplaySupervisor.cs [hunk 1]
```diff
@@ -41,7 +41,10 @@ public class NotificationsDisplaySupervisor : IDisposable
 
         public void DisplayNotification(INotification notification)
         {
-            Dispatch(() => InternalDisplayNotification(notification));
+            if (_dispatcher.CheckAccess())
+                InternalDisplayNotification(notification);
```
Comment (alex-g-axion) [inline Src/ToastNotifications/Display/NotificationsDisplaySupervisor.cs:45]: This is the fix:  Dispatch() would run the message pump resulting in a recursive call.

File: Src/ToastNotifications/Display/NotificationsDisplaySupervisor.cs [hunk 2]
```diff
@@ -56,12 +59,15 @@ private void InternalDisplayNotification(INotification notification)
 
         private void Close(INotification notification)
         {
-            Dispatch(() => InternalClose(notification));
+            if (_dispatcher.CheckAccess())
```
Comment (alex-g-axion) [inline Src/ToastNotifications/Display/NotificationsDisplaySupervisor.cs:62]: Same issue

File: Src/ToastNotifications/Display/NotificationsWindow.xaml.cs [hunk 3]
```diff
@@ -57,18 +57,15 @@ public void CloseNotification(NotificationDisplayPart notification)
...
-                {
-                    this.Visibility = Visibility.Collapsed;
-                }
-                else
-                {
-                    this.Visibility = Visibility.Visible;
-                }
-    
-            })), DispatcherPriority.Render);
+                Dispatcher.Invoke(new Action(RecomputeLayout), DispatcherPriority.Render);
```
Comment (alex-g-axion) [inline Src/ToastNotifications/Display/NotificationsWindow.xaml.cs:62]: Same issue

File: Src/ToastNotifications/Display/NotificationsWindow.xaml.cs [hunk 4]
```diff
@@ -57,18 +57,15 @@ public void CloseNotification(NotificationDisplayPart notification)
...
+                return;
+            }
+
+            this.Visibility = NotificationsList.GetItemCount() == 0
+                ? Visibility.Collapsed
+                : Visibility.Visible;
```
Comment (jneedham-axion) [inline Src/ToastNotifications/Display/NotificationsWindow.xaml.cs:68]: This makes sense, but this was also intentional?

**Review by alisonaquinas — APPROVED** [review]
I think this looks good! Thanks for tracking it down!

**Review by chord-axion — APPROVED** [review]

**Pull request merged**